### DISCUSSION TITLE
Windows 10 ARM build 21354 preview fragile boot

### DISCUSSION THREAD

**Discussion by Digital-Nomad** [utmapp/UTM#2621] [UTM Help (macOS)]
** The issue**
I am having issues where whenever I reboot after installation of the Windows 10 ARM virtual machine it hangs on the black screen(the one after the loading spinner). Windows 10 ARM virtualization seems to be very fragile from what I see. 

**Configuration**
* UTM Version: 2.1.1
* OS Version: Windows 10 ARM preview 21354 VHDX image
* Intel or Apple Silicon? M1

**debug.log**
```
Running:  -L /Applications/UTM.app/Contents/Resources/qemu -S -qmp tcp:127.0.0.1:4000,server,nowait -vga none -spice port=4001,addr=127.0.0.1,disable-ticketing,image-compression=off,playback-compression=off,streaming-video=off -device virtio-ramfb -cpu cortex-a72 -smp cpus=8,sockets=1,cores=8,threads=1 -machine virt,highmem=off -accel hvf -accel tcg,tb-size=1024 -bios /Applications/UTM.app/Contents/Resources/qemu/edk2-aarch64-code.fd -boot menu=on -m 4096 -device intel-hda -device hda-duplex -name Win10ARM -device qemu-xhci,id=usb-bus -device usb-tablet,bus=usb-bus.0 -device usb-mouse,bus=usb-bus.0 -device usb-kbd,bus=usb-bus.0 -device ich9-usb-ehci1,id=usb-controller-0 -device ich9-usb-uhci1,masterbus=usb-controller-0.0,firstport=0,multifunction=on -device ich9-usb-uhci2,masterbus=usb-controller-0.0,firstport=2,multifunction=on -device ich9-usb-uhci3,masterbus=usb-controller-0.0,firstport=4,multifunction=on -chardev spicevmc,name=usbredir,id=usbredirchardev0 -device usb-redir,chardev=usbredirchardev0,id=usbredirdev0,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev1 -device usb-redir,chardev=usbredirchardev1,id=usbredirdev1,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev2 -device usb-redir,chardev=usbredirchardev2,id=usbredirdev2,bus=usb-controller-0.0 -device nvme,drive=drive0,serial=drive0,bootindex=0 -drive if=none,media=disk,id=drive0,file=/Users/simon/Library/Containers/com.utmapp.UTM/Data/Documents/Win10ARM.utm/Images/Windows10_InsiderPreview_Client_ARM64_en-us_21354.VHDX,cache=writethrough -device usb-storage,drive=drive1,removable=true,bootindex=1 -drive if=none,media=cdrom,id=drive1 -device virtio-net-pci,mac=BE:5E:A2:53:FC:4B,netdev=net0 -netdev user,id=net0 -device virtio-serial -device virtserialport,chardev=vdagent,name=com.redhat.spice.0 -chardev spicevmc,id=vdagent,debug=0,name=vdagent -uuid A12F3D46-88CC-4273-AB11-4EF70B09E312 -rtc base=localtime
qemu-aarch64-softmmu: -spice port=4001,addr=127.0.0.1,disable-ticketing,image-compression=off,playback-compression=off,streaming-video=off: warning: short-form boolean option 'disable-ticketing' deprecated
Please use disable-ticketing=on instead
```

**Comment by conath** [comment]
I have had a similar issue when installing Windows 10 in UTM on my iPad. The problem is that Windows 10 setup requires a working automatic DNS to finish installing. This is not the case in UTM unfortunately. Therefore the setup will hang at the step when it waits for the network to become available.

There are two ways to work around this:
1. Enter Windows Audit mode with shift-ctrl-F3 and create user, set OOBE registry keys from there ([tutorial](https://www.youtube.com/watch?v=lhoMdp9qb6Q))
2. Wait for a long time at the black/spinner screen until setup times out and you can proceed offline (this seems to be buggy from Windows setup side so not recommended)
3. Set the DNS manually using Command Prompt. The problem is that you can't always access CMD after setup restarts the computer, so this might not work for all stages of the setup.

**Reply by Digital-Nomad** [reply]
I don't think that's what it is. It completes the setup and boots to the desktop but when I reboot it, it hangs after the loading spinner. It's probably something else.
